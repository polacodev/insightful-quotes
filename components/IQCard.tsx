import React from "react";
import { View, Text } from "react-native";

import IQIcon from "@/components/IQIcon";
import { Quote } from "@/interfaces/quote";

interface IQCardProps {
  quote: Quote;
};

const IQCard = ({ quote }: IQCardProps) => {
  return (
    <View className=" w-[100%] justify-center items-center pt-6 pb-6">
      <View className="w-[85%] bg-[#ffffff] rounded-xl p-5 gap-4 border border-gray-200">
        <View className="items-center gap-4">
          <Text style={{ fontFamily: 'NunitoSans-Variable', fontSize: 16 }} className="text-center">"{quote.body}"</Text>
          <Text style={{ fontFamily: 'NunitoSans-Medium', fontSize: 18 }}>â€”{quote?.author}</Text>
          {quote.tags.length > 0 && <Text style={{ fontFamily: 'NunitoSans-Italic', fontSize: 12 }}>{quote.tags.join(", ")}</Text>}
        </View>
        <View className="flex-row justify-center items-center gap-4 ">
          <View className="flex-col justify-center items-center">
            <IQIcon iconName="heart-dislike-outline" color="#a11313" />
            <Text>
              {quote.downvotes_count}
            </Text>
          </View>
          <View className="flex-col justify-center items-center">
            <IQIcon iconName="heart-outline" color="#ff2323" />
            <Text>
              {quote.upvotes_count}
            </Text>
          </View>
          <View className="flex-col justify-center items-center">
            <IQIcon iconName="star-outline" color="#dbd417" />
            <Text>
              {quote.favorites_count}
            </Text>
          </View>
        </View>
      </View>
    </View>
  );
};

export default IQCard;
