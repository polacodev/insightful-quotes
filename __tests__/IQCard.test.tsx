import { render } from "@testing-library/react-native";

import IQCard from "@/components/IQCard";

jest.mock("expo-font", () => ({
  useFonts: jest.fn(),
  isLoaded: jest.fn(),
  loadAsync: jest.fn(),
}));

describe("<IQCard />", () => {
  const quote = {
    id: 590,
    dialogue: false,
    private: false,
    tags: [
      "funny",
      "work",
      "british",
      "writer"
    ],
    url: "https://favqs.com/quotes/douglas-adams/590-i-love-deadline-",
    favorites_count: 2,
    upvotes_count: 0,
    downvotes_count: 0,
    author: "Douglas Adams",
    author_permalink: "douglas-adams",
    body: "I love deadlines. I like the whooshing sound they make as they fly by."
  }

  test("IQCard renders correctly the tags and Snapshot", () => {
    const { getByText, toJSON } = render(<IQCard quote={quote} />);

    const tags = getByText("funny, work, british, writer");

    expect(tags).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
