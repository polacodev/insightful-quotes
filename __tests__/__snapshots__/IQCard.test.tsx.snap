// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IQCard /> IQCard renders correctly the tags and Snapshot 1`] = `
<View
  className=" w-[100%] justify-center items-center pt-6 pb-6"
>
  <View
    className="w-[85%] bg-[#ffffff] dark:bg-[#181a1b] rounded-xl p-5 gap-4 border border-gray-200 dark:border-[#414749]"
  >
    <View
      className="items-center gap-4"
    >
      <Text
        className="text-center dark:text-[#ffffff]"
        style={
          {
            "fontFamily": "NunitoSans-Variable",
            "fontSize": 16,
          }
        }
      >
        "
        I love deadlines. I like the whooshing sound they make as they fly by.
        "
      </Text>
      <Text
        className="dark:text-[#dcdcdc]"
        style={
          {
            "fontFamily": "NunitoSans-Medium",
            "fontSize": 18,
          }
        }
      >
        â€”
        Douglas Adams
      </Text>
      <Text
        className="dark:text-[#dcdcdc]"
        style={
          {
            "fontFamily": "NunitoSans-Italic",
            "fontSize": 12,
          }
        }
      >
        funny, work, british, writer
      </Text>
    </View>
    <View
      className="flex-row justify-center items-center gap-4 "
    >
      <View
        className="flex-col justify-center items-center"
      >
        <Text />
        <Text
          className="dark:text-[#dcdcdc]"
        >
          0
        </Text>
      </View>
      <View
        className="flex-col justify-center items-center"
      >
        <Text />
        <Text
          className="dark:text-[#dcdcdc]"
        >
          0
        </Text>
      </View>
      <View
        className="flex-col justify-center items-center"
      >
        <Text />
        <Text
          className="dark:text-[#dcdcdc]"
        >
          2
        </Text>
      </View>
    </View>
  </View>
</View>
`;
